### DISCUSSION TITLE
ISO 8601

### DISCUSSION THREAD

**Discussion by drmax24** [rebing/graphql-laravel#797] [General]
I'm a bit confused. Our front end developers suggest that we should use ISO 8601 date format.
"Y-m-d\TH:i:s"
2021-06-19T20:00:00
not
2021-06-19 20:00:00
After some research, I find it difficult to be implemented easily.

I can write a mutator for each date attribute for each model.
I can write a resolver for each attribute in graphql type.

Having ~50 different graphql types and more date attributes it is a lot of dumb code + a big room for a mistake. Sooner or later there can be a mistake and some date would be formatted as 2021-06-19 20:00:00. Losing coherence. May be you can have an idea for simple solution? One that seems most efficient for me is to run regexp through entire server response before it is sent to client (obviously it has it's own disadvantages).

**Comment by mfn** [comment]
You can solve this by using the following approach, _assuming_: you're using Eloquent and are returning Carbon instances (or a compatible representation):
- adding a custom scalar time (e.g. `DateTime`), please see https://github.com/rebing/graphql-laravel#scalar-types
- change all your GraphQL fields to return this type instead of, presumably, string

I suggest to take inspiration from https://github.com/nuwave/lighthouse/blob/94084e7ef882311ce0cb5b413ee9a56c696e0759/src/Schema/Types/Scalars/DateTimeTz.php#L7 (also check the parent `DateScalar` for the big picture).

And yes, I'm also with your frontend team regarding this. It's more or less unambiguous, plus includes the time zone
